lib/puppetserver/ca/action/prune.rb

```diff
@@ -187,6 +196,34 @@ def prune_using_certname(crl, key, inventory_file, cadir, certnames)
           end
         end
 
+        def prune_expired (crl, key, inventory_file, cadir)
+          serialnumbers = []
+          signed_dir = "#{cadir}/signed"
+          @logger.debug("Checking inventory file #{inventory_file} for expired entries") if @logger.debug?
+          errors = FileSystem.validate_file_paths(inventory_file)
+          if errors.empty?
+            File.open(inventory_file).each_line do |line|
+              if line.match(/\/CN=.*$/) && line.split.length == 4
+                not_after = line.split[2]
+                if Time.strptime(not_after,"%Y-%m-%dT%H:%M:%SUTC") && Time.parse(not_after) < Time.now
```

Comment:
I thing strptime is part of DateTime, not Time, but I'm also not sure we really need to do this?  What was this check trying to accomplish?

Reply:
`Time.parse` will throw an exception if the string is not a parseable format, so we may want to handle that here and skip it if it doesn't parse correctly.

Reply:
strptime was added to make sure it confirms to the correct pattern, otherwise skip it. When I was testing it, probably the line was skipped for some other reason i need to go back and check.

Reply:
I think it should be okay not to worry about this.  `Time.parse` can handle a variety of formats, and it results in a `Time` object that doesn't really matter what original form the string was in.  As long as it can parse, it can compare against `Time.now`.

Reply:
Removed strptime and updated to catch the error. Also updated spec test.